gr-utils/python/blocktool/core/parseheader.py

```diff
@@ -0,0 +1,281 @@
+# any later version.
+#
+# GNU Radio is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with GNU Radio; see the file COPYING.  If not, write to
+# the Free Software Foundation, Inc., 51 Franklin Street,
+# Boston, MA 02110-1301, USA.
+#
+""" Module to generate AST for the headers and parse it """
+
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import os
+import re
+import codecs
+import logging
+
+import click
+
+from pygccxml import parser
+from pygccxml import declarations
+from pygccxml import utils
+
+from blocktool.core.base import BlockToolException, BlockTool
+from blocktool.core.iosignature import io_signature
+from blocktool.core import Constants
+
+LOGGER = logging.getLogger(__name__)
+
+
+class BlockHeaderParser(BlockTool):
+    """
+    : Single argument required: file_path
+    file_path: enter path for the header block in any of GNU Radio module
+        : returns the parsed header data in python dict
+        : return dict keys: namespace, class, io_signature, make,
+                       properties, methods
+    : Can be used as an CLI command or an extenal API
+    """
+    name = 'Parse Header'
+    description = 'Create a parsed output from a block header file'
+    module_types = []
+    header_list = []
+    current_folder = os.path.abspath(os.path.dirname(__file__))
+    target_dir = os.path.abspath(os.path.join(current_folder,
+                                              '..', '..', '..', '..'))
+    for list_dir in os.listdir(target_dir):
+        if list_dir.startswith(Constants.GR):
+            list_dir = list_dir.split('-')[-1]
+            module_types.append(list_dir)
+    module_types.remove(Constants.UTILS)
+    module_types = tuple(module_types)
+    parsed_data = {}
+
+    def __init__(self, file_path=None, **kwargs):
+        """ __init__ """
+        BlockTool.__init__(self, **kwargs)
+        try:
+            open(file_path, 'r')
+        except TypeError:
+            raise TypeError
+        file_path = os.path.abspath(file_path)
+        self.info['target_file'] = file_path
+        self.info['modname'] = os.path.basename(os.path.dirname(
+            os.path.dirname(os.path.dirname(os.path.dirname(file_path)))))
+        self.info['filename'] = os.path.basename(file_path)
+        self.info['target_dir'] = os.path.dirname(file_path)
+        self.info['impl_dir'] = os.path.abspath(os.path.join(file_path,
+                                                             '..', '..', '..', '..', 'lib'))
+        self.info['impl_file'] = os.path.join(self.info['impl_dir'],
+                                              self.info['filename'].split('.')[0]+'_impl.cc')
```

Comment:
why aren't these defined as attributes of the instance? want benefit does "info" have